src/type.zig

```diff
@@ -229,8 +229,9 @@ pub const Type = extern union {
             .ErrorSet => {
                 if (a.tag() == .anyerror) {
                     if (b.tag() == .anyerror)
-                        return true;
-                    return false;
+                        return true
+                    else
+                        return false;
```

Comment:
This still doesn't detect `b == .anyerror and a != .anyerror` and now it is hitting the unreachable.